src/sssd.py

```diff
@@ -0,0 +1,278 @@
+    """
+    packages = ["sssd", "sssd-ldap"]
+    try:
+        apt.update()
+        _logger.info("installing packages %s with apt", packages)
+        apt.add_package(packages)
+    except (apt.PackageNotFoundError, apt.PackageError) as e:
+        raise SSSDOpsError(f"failed to install packages {packages}. reason: {e}")
+
+    _logger.info("seeding initial sssd configuration file at %s", SSSD_CONFIG_FILE)
+    with edit() as config:
+        config["sssd"] = {"config_file_version": "2"}
+
+
+def remove() -> None:
+    """Remove SSSD packages."""
+    packages = ["sssd", "sssd-ldap"]
+    try:
+        _logger.info("removing packages %s with apt", packages)
+        apt.remove_package(packages)
+    except apt.PackageNotFoundError as e:
+        raise SSSDOpsError(f"failed to remove packages {packages}. reason: {e}")
+
+    _logger.info("removing sssd configuration file %s", SSSD_CONFIG_FILE)
+    Path(SSSD_CONFIG_FILE).unlink(missing_ok=True)
+
+
+def version() -> str:
+    """Get the current version of `sssd` installed on the machine.
+
+    Raises:
+        SSSDOpsError: Raised if `sssd` is not installed on the machine.
+    """
+    try:
+        return apt.DebianPackage.from_installed_package("sssd").version.number
+    except apt.PackageNotFoundError as e:
+        raise SSSDOpsError(f"unable to retrieve sssd version. reason {e}")
+
+
+def active() -> bool:
+    """Check if `sssd` service is active."""
+    return systemd.service_running("sssd")
+
+
+def restart() -> None:
+    """Restart `sssd` service."""
+    systemd.service_restart("sssd")
+
+
+def enable() -> None:
+    """Enable `sssd` service."""
+    systemd.service_enable("--now", "sssd")
+
+
+def disable() -> None:
+    """Disable `sssd` service."""
+    systemd.service_disable("--now", "sssd")
+
+
+def read() -> configparser.ConfigParser:
+    """Read the contents of the `sssd.conf` service configuration file."""
+    config = configparser.ConfigParser()
+    config.read(SSSD_CONFIG_FILE)
+    return config
+
+
+@contextmanager
+def edit() -> Iterator[configparser.ConfigParser]:
+    """Edit the `sssd.conf` service configuration file.
+
+    Yields:
+        `ConfigParser` object representing the `sssd.conf` file.
+    """
+    config_file = Path(SSSD_CONFIG_FILE)
+    # Ensure `/etc/sssd` exists. Typically, this dir is created by the `sssd` deb package,
+    # but this method should be impervious to if another process destroys `/etc/sssd`.
+    config_file.parent.mkdir(mode=0o711, parents=True, exist_ok=True)
+    config_file.touch()
+    config_file.chmod(0o600)
+    os.chown(config_file, 0, 0)
+
+    config = read()
+    yield config
+    with config_file.open(mode="wt") as fout:
+        config.write(fout)  # type: ignore
+
+
+def domains() -> list[str]:
+    """Get list of configured sssd domains."""
+    config = read()
+    domains_ = config["sssd"].get("domains", "")
+    return domains_.split(",") if domains_ else []
```

Comment:
Nit: Is the check for `domains_` needed? I'd think `split(",")` would return an empty list for an empty string.

Reply:
@jedel1043 so we do need the check for `domains_`. `.split(",")` doesn't actually return an empty list if there's no comma characters within the string; it just returns the a list of `len` 1 with the unsplit string as the first index: 

```pyi
>>> "nocomma".split(",")
["nocomma"]
```